### ISSUE TITLE
`KBucket.remove_node` doesn't handle nodes in the replacement node list

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/trinity#187]
From discussion here: https://github.com/ethereum/trinity/pull/179#issuecomment-455016419

### What is wrong?

Supposing node `A` is in the list of replacement nodes for some `KBucket` instance, calling `bucket.remove_node(A)` will not remove the node from the replacement nodes list.

### How can it be fixed

Some research should be done to determine what the correct behavior is here.

**Comment by zsaladin** [comment]
Before I left the comment in #179 I sent a pull request same as this issue to a [repository](https://github.com/bmuller/kademlia/pull/66) and it was accepted.

And go-ethereum has same issue. So I proposed. But denied. https://github.com/ethereum/go-ethereum/pull/18476
But as I wrote in PR above I still believe general named function like `remove_node` should work general not in specific cases to avoid mistakes.
